pkg/kubelet/kubelet_node_status.go

```diff
@@ -597,6 +598,27 @@ func (kl *Kubelet) setNodeStatusMachineInfo(node *v1.Node) {
 		}
 		node.Status.Allocatable[k] = value
 	}
+
+	hdlr := kl.containerManager.GetDevicePluginHandler()
+	if hdlr == nil {
+		return
+	}
+
+	for k, v := range hdlr.Devices() {
+		key := v1.ResourceName(v1.ResourceOpaqueIntPrefix + k)
+
+		var n int64
+		n = 0
+
+		for _, d := range v {
+			if d.Health == pluginapi.Unhealthy {
+				continue
+			}
+			n++
+		}
+
+		node.Status.Capacity[key] = *resource.NewQuantity(n, resource.DecimalSI)
+	}
 }
 
```

Comment:
So I guess it is not the ultimate realization yet? I could not see things like allocation and real callback